airflow/providers/google/cloud/hooks/gcs.py

```diff
@@ -506,7 +529,7 @@ def _call_with_retry(f: Callable[[], None]) -> None:
                     continue
 
         client = self.get_conn()
-        bucket = client.bucket(bucket_name)
+        bucket = client.bucket(bucket_name, user_project=user_project)
```

Comment:
You can add support for "requester pays" buckets by specifying `user_project` when creating a bucket object. All updates in this PR are basically to pass `user_project` from the operator level to this bucket object factory method.